gutils/fsys.c

```diff
@@ -1039,6 +1039,11 @@ char* GFileMimeType(const char *path) {
         } else {
             pt = copy(pt + 1);
             int len = strlen(pt);
+
+            // Handle the .. 'directory'. Prevents underflowing below
+            if(len < 1)
```

Comment:
Don't think this is correct, what if you had a file named `foo.`.

Can you just change the below check to `if(len > 0 && pt[len - 1] == '~')`